### PR TITLE
Add Framework Namespaces and Tests

### PR DESCRIPTION
**Pull request by ricardofiorani**
Added Namespaces
Fixed Some Class Names
Added Tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /nbproject/*
 /vendor/*
+/.idea/*
 composer.lock
```
Comment (fabiocarneiro) [inline .gitignore]: missing EOL

File: phpunit.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+        <testsuite name="Cupcake Framework Test Suite">
+            <directory>./test</directory>
+        </testsuite>
+    </testsuites>
+    <filter>
+        <whitelist processUncoveredFilesFromWhitelist="true">
+            <directory suffix=".php">./src</directory>
+        </whitelist>
+    </filter>
+</phpunit>
```
Comment (fabiocarneiro) [inline phpunit.xml]: missing EOL

File: src/Config/ConfigManager.php [hunk 3]
```diff
@@ -23,14 +33,26 @@ function __construct(array $configFiles) {
...
-    function __invoke() {
+    function __invoke()
+    {
         return $this->config;
     }
 
-    function getConfig($node = '') {
+    /**
+     * @param string $node
+     * @return mixed|ConfigManager
```
Comment (fabiocarneiro) [inline src/Config/ConfigManager.php]: mixed?

File: src/Core/CupCake.php [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
-     * @var ServiceManager 
+     * @var ServiceManager
      */
     private $serviceManager;
 
-    function __construct(ConfigManager $config) {
+    /**
+     * @param ConfigManager $config
+     */
+    function __construct(ConfigManager $config)
```
Comment (fabiocarneiro) [inline src/Core/CupCake.php:28]: missing visibility

File: src/Config/ConfigManager.php [hunk 5]
```diff
@@ -23,14 +33,26 @@ function __construct(array $configFiles) {
...
+     */
+    function getConfig($node = '')
```
Comment (fabiocarneiro) [inline src/Config/ConfigManager.php]: missing visibility

File: phpunit.xml (hunk 2 above)
Comment (ricardofiorani) [inline phpunit.xml]: Fixed

File: src/Config/ConfigManager.php [hunk 6]
```diff
@@ -1,14 +1,24 @@
...
-    private $config = array();
+    private $configFiles = [];
+    private $config = [];
 
-    function __construct(array $configFiles) {
+    /**
+     * @param array $configFiles
+     * @throws Exception
+     */
+    function __construct(array $configFiles)
```
Comment (fabiocarneiro) [inline src/Config/ConfigManager.php]: missing visibility

File: src/Core/CupCake.php [hunk 7]
```diff
@@ -30,12 +40,18 @@ function __construct(ConfigManager $config) {
         $this->setDbForPainel();
     }
 
-    function run() {
+    /**
+     * @return mixed
+     */
+    function run()
```
Comment (fabiocarneiro) [inline src/Core/CupCake.php:46]: missing visibility

File: src/Core/CupCake.php [hunk 8]
```diff
@@ -73,26 +95,38 @@ public function getActionParameters(array $parameters) {
...
         }
+
         return 'action' . $action;
     }
 
     /**
      * @return ServiceManager
      */
-    function getServiceManager() {
+    function getServiceManager()
```
Comment (fabiocarneiro) [inline src/Core/CupCake.php:121]: missing visibility

File: src/Core/CupCake.php [hunk 9]
```diff
@@ -100,7 +134,8 @@ function setServiceManager(ServiceManager $serviceManager) {
 
     public $db;
 
-    public function setDbForPainel() {
+    public function setDbForPainel()
```
Comment (fabiocarneiro) [inline src/Core/CupCake.php:138]: Missing docblock

File: src/Mailer/MailerManager.php [hunk 10]
```diff
@@ -9,30 +14,45 @@ class MailerManager {
...
+     */
+    public function __construct(array $config, CupRendererInterface $renderer, $dumpMailOnScreen = false)
+    {
         $this->renderer = $renderer;
         $this->config = $config;
         $this->dumpEmailOnScreen = $dumpMailOnScreen;
     }
 
-    public function enviaEmail($dados, $to, $subject = 'Contato através do site', $viewEmail = 'email/contato') {
+    public function enviaEmail($dados, $to, $subject = 'Contato através do site', $viewEmail = 'email/contato')
```
Comment (fabiocarneiro) [inline src/Mailer/MailerManager.php:38]: missing docblock

File: src/Mailer/MailerManager.php [hunk 11]
```diff
@@ -9,30 +14,45 @@ class MailerManager {
...
+    /**
+     * @var array
+     */
     private $config;
 
-    public function __construct(array $config, CupRenderer $renderer, $dumpMailOnScreen = false) {
+    /**
+     * @param array $config
+     * @param CupRendererInterface $renderer
+     * @param bool|false $dumpMailOnScreen
```
Comment (fabiocarneiro) [inline src/Mailer/MailerManager.php:29]: bool|false?

File: src/Mailer/MailerManager.php [hunk 12]
```diff
@@ -9,30 +14,45 @@ class MailerManager {
...
-    private function getRenderedEmail($view, $dados) {
+    private function getRenderedEmail($view, $dados)
+    {
         $this->getRenderer()->setTemplate($this->template);
+
         return $this->getRenderer()->renderizar($view, $dados, true);
     }
 
-    private function send() {
+    private function send()
```
Comment (fabiocarneiro) [inline src/Mailer/MailerManager.php:54]: missing docblock

add return type or return void

File: src/Mailer/MailerManager.php [hunk 13]
```diff
@@ -9,30 +14,45 @@ class MailerManager {
...
         $this->to = $to;
         $this->subject = $subject;
-        $this->message = $this->getRenderedEmail($viewEmail, array('dados' => $dados, 'subject' => $this->subject));
+        $this->message = $this->getRenderedEmail($viewEmail, ['dados' => $dados, 'subject' => $this->subject]);
+
         return $this->send();
     }
 
-    private function getRenderedEmail($view, $dados) {
+    private function getRenderedEmail($view, $dados)
```
Comment (fabiocarneiro) [inline src/Mailer/MailerManager.php:47]: missing docblock

add return type or return void

File: src/Messenger/FlashMessenger.php [hunk 14]
```diff
@@ -1,43 +1,63 @@
...
+    const TYPE_ERROR = 2;
+
 
     private $sessionId;
 
-    function __construct($sessionId = 'messenger-default') {
+    /**
+     * @param string $sessionId
+     */
+    function __construct($sessionId = 'messenger-default')
```
Comment (fabiocarneiro) [inline src/Messenger/FlashMessenger.php:23]: missing visibility

File: src/Messenger/FlashMessenger.php [hunk 15]
```diff
@@ -1,43 +1,63 @@
...
+     * @param $mensagem
+     */
+    public function adicionarMensagemErro($mensagem)
+    {
         $this->adicionarMensagem($mensagem, 2);
     }
 
-    public function adicionarMensagemSucesso($mensagem) {
+    /**
+     * @param $mensagem
```
Comment (fabiocarneiro) [inline src/Messenger/FlashMessenger.php:39]: missing type

File: src/Messenger/FlashMessenger.php [hunk 16]
```diff
@@ -1,43 +1,63 @@
...
+    function __construct($sessionId = 'messenger-default')
+    {
         $this->sessionId = $sessionId;
         $this->startSession();
     }
 
-    public function adicionarMensagemErro($mensagem) {
+
+    /**
+     * @param $mensagem
```
Comment (fabiocarneiro) [inline src/Messenger/FlashMessenger.php:31]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 17]
```diff
@@ -1,31 +1,45 @@
...
     /**
-     * @var PDO 
+     * @var PDO
      */
     private $db;
     private $baseUrl;
     private $debug;
 
-    function __construct(PDO $db, RequestManager $cpr, $debug = false) {
+    function __construct(PDO $db, RequestManager $cpr, $debug = false)
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:27]: visibility

File: src/ObjectMapper/ObjectMapper.php [hunk 18]
```diff
@@ -92,30 +135,57 @@ public function gerarGaleria($data, $caminho = '', $retorno_unico = false) {
         }
     }
 
-    public function array_to_object($array) {
+    /**
+     * Converts an array to an stdClass Object
+     * @param $array
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:140]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 19]
```diff
@@ -92,30 +135,57 @@ public function gerarGaleria($data, $caminho = '', $retorno_unico = false) {
...
+
             return $obj;
         } else {
             return $array;
         }
     }
 
-    public function retornoRegistroPadrao($tabela, $url = '', $pagina = 1, $qtd_registros = 0, $where_custom = 'where ativo = "Sim"', $campo_ordem = 'ordem', $campo_group = '') {
+    /**
+     * @param $tabela
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:162]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 20]
```diff
@@ -257,7 +328,14 @@ public function retornoRegistroPadrao($tabela, $url = '', $pagina = 1, $qtd_regi
         );
     }
 
-    public function verRegistroPadrao($tabela, $id = 0) {
+    /**
+     * @param $tabela
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:332]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 21]
```diff
@@ -341,29 +419,53 @@ public function verRegistroPadrao($tabela, $id = 0) {
         return $row;
     }
 
-    public function geraUrl($input_str) {
+    /**
+     * @param $input_str
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:423]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 22]
```diff
@@ -341,29 +419,53 @@ public function verRegistroPadrao($tabela, $id = 0) {
...
         $input_str = preg_replace("/[\s-]+/", " ", $input_str);
         $input_str = preg_replace("/[\s_]/", "-", $input_str);
+
         return $input_str;
     }
 
-    public function stringUrlAmigavel($phrase, $maxLength = 50) {
+
+    /**
+     * @param $phrase
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:439]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 23]
```diff
@@ -341,29 +419,53 @@ public function verRegistroPadrao($tabela, $id = 0) {
...
+     * @return string
+     */
+    public function removeAcentos($var)
+    {
         return $this->normaliza($var);
     }
 
-    public function normaliza($string) {
+    /**
+     * @param $string
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:464]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 24]
```diff
@@ -371,34 +473,49 @@ public function normaliza($string) {
         $string = utf8_decode($string);
         $string = strtr($string, utf8_decode($a), $b);
         $string = strtolower($string);
+
         return utf8_encode($string);
     }
 
-    public function resumirStr($texto, $n = 20) {
+    /**
+     * @param $texto
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:481]: missing type

File: src/ObjectMapper/ObjectMapper.php [hunk 25]
```diff
@@ -371,34 +473,49 @@ public function normaliza($string) {
...
+                }
             }
         }
+
         return $texto;
     }
 
-    public function getTextoGeral($cod) {
+    /**
+     * @param $cod
```
Comment (fabiocarneiro) [inline src/ObjectMapper/ObjectMapper.php:508]: missing type

File: src/Renderer/CupRenderer.php [hunk 26]
```diff
@@ -15,11 +22,18 @@ class CupRenderer {
...
     /**
      *
-     * @var UrlGenerator 
+     * @var UrlGenerator
      */
     public $urlGenerator;
 
-    function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $urlGenerator) {
+    /**
+     * @param $pastaTemplates
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:30]: missing type

File: src/Renderer/CupRenderer.php [hunk 27]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
         $this->urlGenerator = $urlGenerator;
     }
 
+
     /**
-     * 
      * @return UrlGenerator
      */
-    function getUrlGenerator() {
+    function getUrlGenerator()
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:48]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 28]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         return $this->urlGenerator;
     }
 
-    function getPastaTemplates() {
+    /**
+     * @return mixed
+     */
+    function getPastaTemplates()
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:56]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 29]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         return $this->pastaTemplates;
     }
 
-    function getPastaViews() {
+    /**
+     * @return mixed
+     */
+    function getPastaViews()
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:64]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 30]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         return $this->pastaViews;
     }
 
-    function getTemplate() {
+    /**
+     * @return mixed
+     */
+    function getTemplate()
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:72]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 31]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         return $this->template;
     }
 
-    function getTituloSite() {
+    /**
+     * @return mixed
+     */
+    function getTituloSite()
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:80]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 32]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         return $this->tituloSite;
     }
 
-    function setPastaTemplates($pastaTemplates) {
+    /**
+     * @param $pastaTemplates
+     */
+    function setPastaTemplates($pastaTemplates)
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:88]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 33]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         $this->pastaTemplates = $pastaTemplates;
     }
 
-    function setPastaViews($pastaViews) {
+    /**
+     * @param $pastaViews
+     */
+    function setPastaViews($pastaViews)
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:96]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 34]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         $this->pastaViews = $pastaViews;
     }
 
-    function setTemplate($template) {
+    /**
+     * @param $template
+     */
+    function setTemplate($template)
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:104]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 35]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         $this->template = $template;
     }
 
-    function setTituloSite($tituloSite) {
+    /**
+     * @param $tituloSite
+     */
+    function setTituloSite($tituloSite)
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:112]: missing visibility

File: src/Renderer/CupRenderer.php [hunk 36]
```diff
@@ -27,47 +41,88 @@ function __construct($pastaTemplates, $pastaViews, $tituloSite, UrlGenerator $ur
...
+    {
         $this->tituloSite = $tituloSite;
     }
 
-    public function renderizar($nomeView, array $variaveis = array(), $retornar = false) {
+    /**
+     * @param $nomeView
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:118]: missing type

File: src/Renderer/CupRenderer.php [hunk 37]
```diff
@@ -84,7 +139,15 @@ public function renderizar($nomeView, array $variaveis = array(), $retornar = fa
         return $this->renderView($template, $variaveis, $retornar);
     }
 
-    public function renderizarParcial($nomeView, array $variaveis = array(), $retornar = false) {
+    /**
+     * @param $nomeView
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:143]: missing type

File: src/Renderer/CupRenderer.php [hunk 38]
```diff
@@ -92,10 +155,18 @@ public function renderizarParcial($nomeView, array $variaveis = array(), $retorn
                 throw new Exception(sprintf('A view %s não foi encontrada', $view));
             }
         }
+
         return $this->renderView($view, $variaveis, $retornar);
     }
 
-    public function renderView($arquivoParaRenderizar, $variaveis = array(), $retornar = false) {
+    /**
+     * @param $arquivoParaRenderizar
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:163]: missing type

File: src/Renderer/CupRenderer.php [hunk 39]
```diff
@@ -92,10 +155,18 @@ public function renderizarParcial($nomeView, array $variaveis = array(), $retorn
...
+     * @param array $variaveis
+     * @param bool|false $retornar
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:165]: bool|false?

File: src/Renderer/CupRenderer.php [hunk 40]
```diff
@@ -116,46 +187,77 @@ public function renderView($arquivoParaRenderizar, $variaveis = array(), $retorn
...
+            case 11:
+                return "Novembro";
+            case 12:
+                return "Dezembro";
         }
     }
 
-    public function hideEmailFromString($string) {
+    /**
+     * @param $string
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:240]: missing type

File: src/Renderer/CupRenderer.php [hunk 41]
```diff
@@ -116,46 +187,77 @@ public function renderView($arquivoParaRenderizar, $variaveis = array(), $retorn
...
         $pedacos = explode('@', $string);
         $domain = '@' . end($pedacos);
         $stringOcultada = substr_replace(reset($pedacos), '*********', 5);
+
         return $stringOcultada . $domain;
     }
 
-    public function hideNameFromString($string) {
+    /**
+     * @param $string
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:256]: missing type

File: src/Renderer/CupRenderer.php [hunk 42]
```diff
@@ -165,23 +267,37 @@ public function hideNameFromString($string) {
...
         if (empty($string)) {
             return;
         }
+
         return substr_replace($string, str_repeat('*', strlen($string) - 5), 3) . substr($string, -2);
     }
 
-    public function hideString($string, $size = 5) {
+    /**
+     * @param $string
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:288]: missing type

File: src/Renderer/CupRenderer.php [hunk 43]
```diff
@@ -165,23 +267,37 @@ public function hideNameFromString($string) {
         foreach ($pedacos as $p) {
             $nome .= ' ' . substr($p, 0, 1) . str_repeat('*', strlen($p) - 3) . substr($p, -2);
         }
+
         return $nome;
     }
 
-    public function hideTelephoneNumberFromString($string) {
+    /**
+     * @param $string
```
Comment (fabiocarneiro) [inline src/Renderer/CupRenderer.php:275]: missing type

File: src/Renderer/CupRendererInterface.php [hunk 44]
```diff
@@ -0,0 +1,27 @@
...
+interface CupRendererInterface
+{
+    /**
+     * @param $template
+     * @return mixed
+     */
+    public function setTemplate($template);
+
+    /**
+     * @param $nomeView
```
Comment (fabiocarneiro) [inline src/Renderer/CupRendererInterface.php:21]: missing type

File: src/Renderer/CupRendererInterface.php [hunk 45]
```diff
@@ -0,0 +1,27 @@
...
+     * @param array $variaveis
+     * @param bool|false $retornar
+     * @return mixed
+     */
+    public function renderizar($nomeView, array $variaveis = [], $retornar = false);
+}
```
Comment (fabiocarneiro) [inline src/Renderer/CupRendererInterface.php:27]: EOF EOL

File: src/Request/RequestManager.php [hunk 46]
```diff
@@ -16,26 +18,44 @@ class RequestManager {
      */
     private $context;
 
-    function __construct() {
+    function __construct()
```
Comment (fabiocarneiro) [inline src/Request/RequestManager.php:21]: missing visibility

File: src/Request/RequestManager.php [hunk 47]
```diff
@@ -16,26 +18,44 @@ class RequestManager {
...
+     * @return string
+     */
+    public function getBaseUrl()
+    {
         return $this->getContext()->getBaseUrl();
     }
 
-    public function redirect($url) {
+    /**
+     * @param $url
```
Comment (fabiocarneiro) [inline src/Request/RequestManager.php:44]: missing type

File: src/Service/Exception/ContainerException.php [hunk 48]
```diff
@@ -1,10 +1,14 @@
...
 use Interop\Container\Exception\ContainerException as ContainerExceptionInterface;
 
 /**
  * @author Ricardo Fiorani
  */
-class ContainerException extends Exception implements ContainerExceptionInterface {
-    
+class ContainerException extends Exception implements ContainerExceptionInterface
+{
+
```
Comment (fabiocarneiro) [inline src/Service/Exception/ContainerException.php:13]: Remove this line

File: src/Service/Exception/NotFoundException.php [hunk 49]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Cupcake\Service\Exception;
+
+
+/**
+ * @author Ricardo Fiorani
+ */
+class NotFoundException extends ContainerException
+{
+
```
Comment (fabiocarneiro) [inline src/Service/Exception/NotFoundException.php:11]: Remove this line

File: src/Url/UrlGenerator.php [hunk 50]
```diff
@@ -1,17 +1,22 @@
...
 
     /**
      *
-     * @var RequestManager 
+     * @var RequestManager
      */
     private $request;
 
-    function __construct(RequestManager $cpr) {
+    function __construct(RequestManager $cpr)
```
Comment (fabiocarneiro) [inline src/Url/UrlGenerator.php:18]: visibility

File: src/Url/factory/UrlGeneratorFactory.php [hunk 51]
```diff
@@ -1,11 +1,21 @@
...
-class UrlGeneratorFactory {
+class UrlGeneratorFactory
+{
 
-    function __invoke(ServiceManager $serviceManager) {
+    /**
+     * @param ServiceManager $serviceManager
+     * @return UrlGenerator
+     */
+    function __invoke(ServiceManager $serviceManager)
```
Comment (fabiocarneiro) [inline src/Url/factory/UrlGeneratorFactory.php:17]: visibility

File: src/Validator/AbstractValidator.php [hunk 52]
```diff
@@ -1,42 +1,55 @@
...
+    public function __construct(FlashMessenger $messenger)
+    {
         $this->setMessenger($messenger);
     }
 
-    function getMessenger() {
+    /**
+     * @return FlashMessenger
+     */
+    function getMessenger()
```
Comment (fabiocarneiro) [inline src/Validator/AbstractValidator.php:29]: visibility

**Pull request merged**